internal/controller/kustomization_deletion_policy_test.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/utils/ptr"
+	"testing"
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/testserver"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1"
+)
+
+func TestKustomizationReconciler_DeletionPolicyDelete(t *testing.T) {
```

Comment:
Please test also `MirrorPrune` with the different possible values for `prune`.

Reply:
@matheuscscp, will do! Are there any recommendations or established patterns in this project to avoid so much duplicated test code? If not, I am pretty good at copy-modify. 😉

Reply:
We always try to do the usual table testing widespread in Go projects. However if the test cases are too different then it's OK to have separate `func TestXXX` tests for them. In this specific case you could for example group together the cases where deletion is expected, and make another group where orphaning is expected.

Reply:
Thanks! Do you have a good example of how you write table-driven tests in Flux projects? Assuming there are multiple ways to do this.

Reply:
Thanks! I have updated the tests now. I have some issues running the tests on my workstation (unrelated to this change). I get the following error:

````
panic: Failed to create Vault instance: could not connect to docker: reached retry deadline: Get "http://127.0.0.1:32769/v1/sys/init": dial tcp 127.0.0.1:32769: connect: connection refused
````

So would it be possible to approve workflows to check the tests (and CI in general)?

I also want to rebase this PR before merge. But I'll wait until it's "approved" (almost ready to merge).